### PR TITLE
dark mode: improve visuals of time series plugin

### PR DESCRIPTION
**Pull request by stephanwlee**
This change improves how TimeSeries dashboard looks under the dark mode.
It largely looks okay but the gpu line chart is a bit broken under it.

Do note that some of the color change will cause minor screenshot test failures.
For instance, we were using white + opacity for border color on TimeSeries
runs table which had to change to a opaque color.

### CODE REVIEW

File: tensorboard/webapp/metrics/views/metrics_container.ts [hunk 1]
```diff
@@ -17,10 +17,12 @@ import {ChangeDetectionStrategy, Component} from '@angular/core';
 @Component({
   selector: 'metrics-dashboard',
   template: `
-    <nav>
-      <runs-selector></runs-selector>
-    </nav>
-    <metrics-main-view></metrics-main-view>
+    <div class="plugin-view">
```
Comment (psybuzz) [inline tensorboard/webapp/metrics/views/metrics_container.ts:20]: I'm not seeing why this extra wrapper was necessary.  Can we drop it, or if not, could you add a comment explaining why it's needed?

File: tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.scss [hunk 2]
```diff
@@ -51,6 +51,10 @@ p {
   display: flex;
   justify-content: space-between;
   padding: 2px 12px;
+
+  @include tb-dark-theme {
+    background-color: #f5f6f722;
```
Comment (psybuzz) [inline tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.scss:56]: This isn't supposed to be a transparent element.  If we want to make it clear from reading that it's the same hue/saturation with higher brightness, consider hsl()?

File: tensorboard/webapp/metrics/views/metrics_container.scss [hunk 3]
```diff
@@ -24,6 +24,11 @@ nav {
   border-right: 1px solid mat-color($tb-foreground, border);
   flex: none;
   width: 340px;
+
+  @include tb-dark-theme {
+    background-color: map-get($tb-dark-background, background);
+    border-right-color: mat-color($tb-dark-foreground, border);
+  }
```
Comment (psybuzz) [inline tensorboard/webapp/metrics/views/metrics_container.scss:31]: To check my understanding, is this equivalent to
```
nav {
  @include tb-theme-background-prop(background-color, background);
  @include tb-theme-foreground-prop(border-right, border, 1px solid);
```
?

File: tensorboard/webapp/theme/_tb_theme.template.scss [hunk 4]
```diff
@@ -68,12 +92,101 @@ $tb-dark-background: map-get($tb-dark-theme, background);
...
+ *   // :host-context(body.dark-mode) div[ngcomponentweirdid] {
+ *   //   border-left: 1px solid map_get($tb-dark-foreground, border);
+ *   // }
+ *   //
+ *   // Third argument is optional. In the example below, `color` only takes
+ *   // color value so it is not needed.
+ *   @include tb-theme-foreground-prop(color, secondary-text);
+ * }
+ */
+@mixin tb-theme-foreground-prop($property, $name, $value: null) {
```
Comment (psybuzz) [inline tensorboard/webapp/theme/_tb_theme.template.scss:142]: The example mentioned is great.

Upon first glance, I think there is a little bit of confusion between these param names.  Wdyt about giving these params more specific names or doing a JSDoc-like comment, something along the lines of:

- $property -> $css-property
- $name -> $tb-variable  (must be a key in $tb-theme)
- $value -> $css-value-prefix

?

**Review by psybuzz — APPROVED** [review]
Nice!

File: tensorboard/webapp/metrics/views/metrics_container.ts (hunk 1 above)
Comment (stephanwlee) [reply tensorboard/webapp/metrics/views/metrics_container.ts]: Removed. It was needed at one point but now can be removed.

File: tensorboard/webapp/metrics/views/metrics_container.scss (hunk 3 above)
Comment (stephanwlee) [reply tensorboard/webapp/metrics/views/metrics_container.scss:31]: Correct; I still cannot make up my mind on which pattern is better when there are more than N *-props call but this is the pattern that I wanted to include in the code base for reference.

File: tensorboard/webapp/theme/_tb_theme.template.scss (hunk 4 above)
Comment (stephanwlee) [reply tensorboard/webapp/theme/_tb_theme.template.scss]: Done.

**Pull request merged**